psst-core/src/session/client_token.rs

```diff
@@ -0,0 +1,252 @@
+// Ported from librespot
```

Comment:
It's not great to have this static code ported over, but is this a full file ported over or just a piece?